engine/apps/api/serializers/alert_receive_channel.py

```diff
@@ -170,18 +167,38 @@ def _create_custom_labels(organization: Organization, labels: AlertGroupCustomLa
         LabelKeyCache.objects.bulk_create(label_keys, ignore_conflicts=True, batch_size=5000)
         LabelValueCache.objects.bulk_create(label_values, ignore_conflicts=True, batch_size=5000)
 
+    @staticmethod
+    def _save_static_labels_as_integration_labels(instance: AlertReceiveChannel, labels: AlertGroupCustomLabelsAPI):
+        labels_associations_to_create = []
+        labels_copy = labels[:]
```

Comment:
why do we need to make a copy of the list + do `labels.remove` further down?

Reply:
removing items during the iteration through the original lists causes some issues since the list's length changes. I see two options to avoid this: iterating through the copy, or add items to remove to a separate list and go through it in another iteration. Lmk if I miss something 🙏